src/lib/storage/value_vector.cpp

```diff
@@ -0,0 +1,207 @@
+}
+
+template <typename T>
+T& ValueVector<T>::at(const ChunkOffset chunk_offset) {
+  return _values.at(chunk_offset);
+}
+
+template <typename T>
+typename ValueVector<T>::iterator ValueVector<T>::begin() noexcept {
+  return _values.begin();
+}
+
+template <typename T>
+typename ValueVector<T>::iterator ValueVector<T>::end() noexcept {
+  return _values.end();
+}
+
+template <typename T>
+typename ValueVector<T>::const_iterator ValueVector<T>::begin() const noexcept {
+  return _values.begin();
+}
+
+template <typename T>
+typename ValueVector<T>::const_iterator ValueVector<T>::end() const noexcept {
+  return _values.end();
+}
+
+template <typename T>
+typename ValueVector<T>::reverse_iterator ValueVector<T>::rbegin() noexcept {
+  return _values.rbegin();
+}
+
+template <typename T>
+typename ValueVector<T>::reverse_iterator ValueVector<T>::rend() noexcept {
+  return _values.rend();
+}
+
+template <typename T>
+typename ValueVector<T>::const_iterator ValueVector<T>::cbegin() const noexcept {
+  return _values.cbegin();
+}
+
+template <typename T>
+typename ValueVector<T>::const_iterator ValueVector<T>::cend() const noexcept {
+  return _values.cend();
+}
+
+template <typename T>
+void ValueVector<T>::erase(iterator start, iterator end) {
+  _values.erase(start, end);
+}
+
+template <typename T>
+T& ValueVector<T>::operator[](const size_t n) {
+  return _values[n];
+}
+
+template <typename T>
+const T& ValueVector<T>::operator[](const size_t n) const {
+  return _values[n];
+}
+
+template <typename T>
+size_t ValueVector<T>::size() const {
+  return _values.size();
+}
+
+template <typename T>
+size_t ValueVector<T>::capacity() const {
+  return _values.capacity();
+}
+
+template <typename T>
+void ValueVector<T>::shrink_to_fit() {
+  _values.shrink_to_fit();
+}
+
+template <typename T>
+PolymorphicAllocator<T> ValueVector<T>::get_allocator() {
+  return _values.get_allocator();
+}
+
+template <typename T>
+void ValueVector<T>::reserve(const size_t n) {
+  _values.reserve(n);
+}
+
+template <typename T>
+const pmr_vector<T>& ValueVector<T>::values() const {
+  return _values;
+}
+
+template <typename T>
+const T* ValueVector<T>::data() const {
+  return &_values[0];
+}
+
+template <typename T>
+size_t ValueVector<T>::data_size() const {
+  return sizeof(*this) + _values.size() * sizeof(typename decltype(_values)::value_type);
+}
+
+// Implementation of ValueVector<FixedString> starts here
+
+void ValueVector<FixedString>::push_back(const std::string& string) {
+  const auto pos = _chars.size();
+  _chars.resize(_chars.size() + _string_length);
+  string.copy(&_chars[pos], _string_length);
+  if (string.size() < _string_length) {
+    std::fill(_chars.begin() + pos + string.size(), _chars.begin() + pos + _string_length, '\0');
+  }
+}
+
+FixedString ValueVector<FixedString>::at(const ChunkOffset chunk_offset) {
+  return FixedString(reinterpret_cast<char*>(&_chars.at(chunk_offset * _string_length)), _string_length);
```

Comment:
why the `reinterpret_cast<char*>`? doesn't `&_chars.at(...)` already return a `char*`?